### ISSUE TITLE
llc4320 ecco portal no mask 

### ISSUE THREAD

**Issue by apatlpo** [MITgcm/xmitgcm#190]
I downloaded data (wind stress llc4320) from the ECCO data portal and would like to read it with llcreader as advertised in this [post](https://medium.com/pangeo/petabytes-of-ocean-data-part-1-nasa-ecco-data-portal-81e3c5e077be).

My understanding is that I need to provide mask information in order to do that.
Could you please confirm?

If this is the case, I have the feeling mask information is not present on the ECCO portal for llc4320.

**Comment by apatlpo** [comment]
pinging again about this ...

**Comment by apatlpo** [comment]
> My understanding is that I need to provide mask information in order to do that.
Could you please confirm?

Answer is yes, you need to have a consolidated zarr archive containing mask_c, mask_w, mask_s variables

**Comment by apatlpo** [comment]
I am still not closing this issue because of the missing mask issue

**Comment by rabernat** [comment]
I am looking at this today. Sorry for the slow response.

**Comment by rabernat** [comment]
p.s. Would be helpful if you could provide more detail about what you have tried, i.e. specific code, and any error messages that came out.

**Comment by apatlpo** [comment]
you're right, sorry about not providing any code. I'll try to do that soon.

If you have time to spend on these issues however, I'd rather you take a look at the other issue I posted yesterday (191), which is much more worrying.

**Comment by apatlpo** [comment]
@menemenlis pushed hFac files back on the ECCO Data portal.

Before closing this issue, I'd be tempted to propose adding one or two sentences to llcreader ["Manual Dataset Creation" doc](https://xmitgcm.readthedocs.io/en/latest/llcreader.html#manual-dataset-creation) referring to the necessity to generate (consolidated) a zarr mask archive.
Do people think this may be useful?

**Comment by rabernat** [comment]
> I'd be tempted to propose adding one or two sentences to llcreader ["Manual Dataset Creation" doc](https://xmitgcm.readthedocs.io/en/latest/llcreader.html#manual-dataset-creation) referring to the necessity to generate (consolidated) a zarr mask archive.
> Do people think this may be useful?

Very useful! Would you be able to make a PR?

One complication is that the masks themselves are stored in a highly compressed format. They are not the raw hFacC.data files, which are extremely inefficient in terms of disk space usage. You can find them here: https://console.cloud.google.com/storage/browser/pangeo-ecco/llc/masks/

It might be wise to put a copy of these in the ecco data portal as well.

**Comment by apatlpo** [comment]
Ok, I can suggest this to Dimitris but I did not figure out how to download the mask from google cloud storage. 

Do you think Dimitris will know how to do this?

If not could you please specify how one can do that?

**Comment by rabernat** [comment]
> Ok, I can suggest this to Dimitris but I did not figure out how to download the mask from google cloud storage.
> 
> Do you think Dimitris will know how to do this?

I made this mask, not Dimitris.

> If not could you please specify how one can do that?

```
gsutil -m cp -r gs://pangeo-ecco/llc/masks/llc_4320_masks.zarr /local/path
```
